python/strings_udf/cpp/src/strings/udf/shim.cu

```diff
@@ -270,3 +271,57 @@ extern "C" __device__ int rstrip(int* nb_retval,
 
   return 0;
 }
+
+extern "C" __device__ int upper(int* nb_retval,
+                                void* udf_str,
+                                void* const* st,
+                                std::int64_t flags_table,
+                                std::int64_t cases_table,
+                                std::int64_t special_table)
+{
+  auto udf_str_ptr = new (udf_str) udf_string;
```

Comment:
This calls the default constructor, but this object is never read anywhere, and is then overwritten on line 297 (`to_upper` returns a `udf_string`). So what is going on here?

Does this want to disappear and line 297 be `*reinterpret_cast<udf_string *>(udf_str) = to_upper(tables, *st_ptr);` ?

Reply:
No the object must be initialized by a constructor in order for it to be valid. Otherwise the `operator=` will crash if the target has garbage in it.

Reply:
Thanks!